Apps/Common/src/AccessManagement/Administration/KeycloakUserAdminDelegate.cs

```diff
@@ -0,0 +1,148 @@
+        public KeycloakUserAdminDelegate(
+            ILogger<KeycloakUserAdminDelegate> logger,
+            IHttpClientService httpClientService,
+            IConfiguration configuration)
+        {
+            this.logger = logger;
+            this.httpClientService = httpClientService!;
+            this.configuration = configuration;
+        }
+
+        /// <inheritdoc/>
+        public UserRepresentation GetUser(Guid userId, JWTModel jwtModel)
+        {
+            this.logger.LogInformation($"Keycloak GetUser : {userId.ToString()}");
+
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public int DeleteUser(Guid userId, JWTModel jwtModel)
+        {
+            this.logger.LogInformation($"Keycloak DeleteUser : {userId.ToString()}");
+
+            Task<int> task = Task.Run(async() => await this.DeleteUserAsync(userId.ToString(), jwtModel).ConfigureAwait(true));
+            task.Wait();
+            if (task.Exception != null)
+            {
+                throw new Exception("KeycloakUserAdminDelegate.DeleteUser() exception", task.Exception);
+            }
+            return task.Result;
+
+        }
+
+        /// <summary>
+        /// Deletes the User account from Keycloak
+        /// </summary>
+        /// <param name="userId">The user id to delete.</param>
+        /// <param name="jwtModel">To get at the base64 access token</param>
+        private async Task<int> DeleteUserAsync(string userId, JWTModel jwtModel)
+        {
+
+            Uri baseUri = new Uri(this.configuration.GetSection(KEYCLOAKADMIN).GetValue<string>(DELETEUSERURL));
+
+            using HttpClient client = this.CreateHttpClient(baseUri, jwtModel.AccessToken!);
+            {
+                HttpResponseMessage response = await client.DeleteAsync(userId).ConfigureAwait(true);
+                if (!response.IsSuccessStatusCode) //Shouuld get a HTTP 204 no content success code from Keycloak API
+                {
+                    string msg = $"Error performing DELETE Request: {userId}, HTTP StatusCode: {response.StatusCode}";
+                    this.logger.LogError(msg);
+
+                    if (response.StatusCode != HttpStatusCode.NotFound) // suppress throwable if resource not found = already deleted
+                    {
+                        throw new HttpRequestException(msg);
+                    }
+                    return -1;
+                }
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Retuns an HttpClient for the AuthService to be invoked.
+        /// </summary>
+        /// <param name="baseUri">The uri of the service endpoint.</param>
+        /// <param name="base64BearerToken">The JSON Web Token.</param>
+        private HttpClient CreateHttpClient(Uri baseUri, string base64BearerToken)
+        {
+            HttpClient client = this.httpClientService.CreateDefaultHttpClient();
+
```

Comment:
NIT/SMELL: Spacing

Reply:
yeah, I don't have my environment setup here for formatting.. sorry.